spring-kafka/src/main/java/org/springframework/kafka/listener/RecordInterceptor.java

```diff
@@ -43,21 +44,38 @@
 	@Nullable
 	ConsumerRecord<K, V> intercept(ConsumerRecord<K, V> record);
```

Comment:
I would say that we don't need this contract at all: it's up to end-user to do anything with that extra `consumer` or not.

More over I see that you have changed `BatchInterceptor` contract for that extra param.
Therefore for consistency this might be deprecated with removal in the future.

But yeah, I'll deffer the final decision to you...

Reply:
Yes, will deprecate - and then we can deprecate the sub-interface when we remove this.